src/mappings/events/balances.ts

```diff
@@ -0,0 +1,159 @@
+import { Balances } from '../../types';
+import { BALANCE_FIELDS } from '../constants';
+import { b64decode } from '../utils';
+import { CosmosEvent } from '@subql/types-cosmos';
+
+interface Attribute {
+  key: string;
+  value: string;
+}
+
+export interface DecodedEvent {
+  type: string;
+  attributes: Attribute[];
+}
+
+export enum Operation {
+  Increment = 'increment',
+  Decrement = 'decrement',
+}
+
+interface BLDTransaction {
+  isBLDTransaction: boolean;
+  amount: string;
+}
+export const balancesEventKit = () => {
+  function getAttributeValue(data: any, key: string) {
+    const attribute = data.attributes.find(
+      (attr: Attribute) => attr.key === key
+    );
+    return attribute ? attribute.value : null;
+  }
+
+  function getData(cosmosEvent: CosmosEvent) {
+    let dataAlreadyDecoded = true;
+    const value = getAttributeValue(cosmosEvent.event, BALANCE_FIELDS.amount);
+
+    if (!value) {
+      dataAlreadyDecoded = false;
+    }
+
+    const data = dataAlreadyDecoded
+      ? cosmosEvent.event
+      : decodeEvent(cosmosEvent);
+
+    return data;
+  }
+
+  function decodeEvent(cosmosEvent: CosmosEvent): DecodedEvent {
+    const { event } = cosmosEvent;
+
+    const decodedData: DecodedEvent = {
+      type: event.type,
+      attributes: [],
+    };
+
+    event.attributes.forEach((attribute) => {
+      const decodedKey = b64decode(attribute.key);
+      const decodedValue = b64decode(attribute.value);
+
+      decodedData.attributes.push({
+        key: decodedKey,
+        value: decodedValue,
+      });
+    });
+
+    return decodedData;
+  }
+
+  async function addressExists(address: string): Promise<boolean> {
+    const balance = await Balances.getByAddress(address);
+
+    if (!balance || balance.length === 0) {
+      return false;
+    }
+    return true;
+  }
+
+  async function createBalancesEntry(address: string) {
+    const newBalance = new Balances(address);
+    newBalance.address = address;
+    newBalance.balance = BigInt(0);
+    newBalance.denom = 'ubld';
+
+    await newBalance.save();
+
+    logger.info(`Created new entry for address: ${address}`);
+  }
+
+  function validateBLDTransaction(amount: string | null): BLDTransaction {
```

Comment:
I think we should not limit just to BLD, and instead let any token balance be indexed.

Reply:
Yeah, we can do that. We'll need to discuss the schema design for that. 
With the current schema, we'll be updating the same amount field for each token.